src/hotspot/share/runtime/atomic.hpp

```diff
@@ -1118,7 +1119,7 @@ inline T Atomic::CmpxchgByteUsingInt::operator()(T volatile* dest,
   uint8_t canon_compare_value = compare_value;
   volatile uint32_t* aligned_dest
     = reinterpret_cast<volatile uint32_t*>(align_down(dest, sizeof(uint32_t)));
-  size_t offset = pointer_delta(dest, aligned_dest, 1);
+  uint32_t offset = checked_cast<uint32_t>(pointer_delta(dest, aligned_dest, 1));
```

Comment:
If this works for all compilers, then that's great. I was a bit concerned that the code in the statement below would cause a warning:
```
(sizeof(uint32_t) - 1 - offset)
```

given that `sizeof` returns a `size_t` and `1` an `int`, but I guess the compilers are smart enough to figure out that they all fit within a uint32_t?

Reply:
gcc doesn't complain about this with -Wsign-conversion, or at least as included with arrayOop.hpp, which was the only goal of this change was to fix the new -Wconversion warning with arrayOop.hpp.